app/models/invoice.rb

```diff
@@ -14,4 +14,20 @@ def self.revenue_by_date(date)
       .sum("invoice_items.unit_price * invoice_items.quantity")
   end
 
+  def self.item_best_day(item_id)
+    unscoped.select("invoices.*, sum(invoice_items.quantity
+                    * invoice_items.unit_price) as revenue")
+      .joins(:transactions, :invoice_items)
+      .merge(Transaction.unscoped.successful)
+      .where("invoice_items.item_id = ?", item_id)
+      .order("revenue desc")
+      .group(:id)
+      .limit(1)
+  end
+
+  def self.dates(item_id)
```

Comment:
This iteration seems a little funky to me, I feel like we should just be able to call .created_at on the object returned by item_best_day?